lib/pgpm/deb/builder.rb

```diff
@@ -0,0 +1,123 @@
+# frozen_string_literal: true
+
+module Pgpm
+  module Deb
+    class Builder
+
+      def initialize(spec)
+        @spec = spec
+        @image_name = "quay.io/qount25/pgpm-debian12"
+        @container_name = "pgpm-debian12_build-#{Time.now.to_i}_#{rand(10000)}"
+      end
+
+      def build
+        puts "build()"
+        prepare
+        generate_deb_src_files
+        create_container
+        run_pbuilder
+        copy_build_from_container
+        cleanup
+      end
+
+      private
+
+      def prepare
+        puts "Preparing build..."
+        puts "  Creating container dir structure..."
+        @pgpm_dir  = Dir.mktmpdir
+        Dir.mkdir "#{@pgpm_dir}/source"
+        Dir.mkdir "#{@pgpm_dir}/out"
+        puts "  Copying #{@spec.package.source.to_s} to #{@pgpm_dir}/source/"
+        FileUtils.copy_entry @spec.package.source.to_s, "#{@pgpm_dir}/source/"
+      end
+
+      def create_container
+        puts "Creating a podman container..."
+        # Check if image exists
+        system("podman image exists #{@image_name}")
+        if $?.to_i > 0 # image doesn't exist -- pull image from a remote repository
+          puts "  Pulling image #{@image_name}..."
+          system("podman pull quay.io/qount25/pgpm-debian12")
+        else
+          puts "  Image #{@image_name} already exists! OK"
+        end
+
+        create_opts = " -v #{@pgpm_dir}:/root/pgpm"
+        create_opts += ":z" if selinux_enabled?
+        create_opts += " --privileged --annotation run.oci.keep_original_groups=1"
+        create_opts += " --name #{@container_name} #{@image_name}"
+
+        puts "  Creating and starting container #{@container_name}"
+        puts "    podman run -dti #{create_opts}"
+        system("podman run -dti #{create_opts}")
```

Comment:
I am wondering what's the rationale behind running the container and attaching to it (and then having to stop it) as opposed to doing one-off runs?